File: static/complicated.js
	
		var drawTimeout = null;	
		document.addEventListener('DOMContentLoaded', function(){
    var v = document.getElementById('v');
    var canvas = document.getElementById('c');
    var context = canvas.getContext('2d');
    var back = document.createElement('canvas');
    var backcontext = back.getContext('2d');
    var cw,ch;

    v.addEventListener('play', function(){
        cw = v.clientWidth;
        ch = v.clientHeight;
        canvas.width = cw;
        canvas.height = ch;
        back.width = cw;
        back.height = ch;
        draw(v,context,backcontext,cw,ch,1);
    },false);

},false);

function draw(v,c,bc,w,h,value) {
    if(v.paused || v.ended) return false;
    // First, draw it into the backing canvas
    bc.drawImage(v,0,0,w,h);
    // Grab the pixel data from the backing canvas
    var idata = bc.getImageData(0,0,w,h);
    var data = idata.data;
    // Loop through the pixels, turning them grayscale
    for(var i = 0; i < data.length; i+=4) {
        var r = data[i];
        var g = data[i+1];
        var b = data[i+2];
        var brightness = (3*r+4*g+b)>>>3;
        data[i] = r * vidData[0] * vidData[1];
        data[i+1] = g * vidData[0] * vidData[2];
        data[i+2] = b * vidData[0] * vidData[3];
    }
    idata.data = data;
    // Draw the pixels onto the visible canvas
    c.putImageData(idata,0,0);
    // Start over!
    drawTimeout = setTimeout(function(){ draw(v,c,bc,w,h,value); }, 0);
}

File: py/complicated.py
drawTimeout = None
vidData = [1,1,1,1]

def lambda_function1():
    def lambda_function2():
        global cw, ch
        cw = v.clientWidth
        ch = v.clientHeight
        canvas.width = cw
        canvas.height = ch 
        back.width = cw
        back.height = ch
        draw(v,context,backcontext,cw,ch,1) 

    v = js.document.getElementById('v')
    canvas = js.document.getElementById('c')
    context = js.canvas.getContext('2d') 
    back = js.document.createElement('canvas')
    backcontext = js.back.getContext('2d')
    cw = None
    ch = None
    js.v.addEventListener('play',lambda_function2,False)

js.document.addEventListener('DOMContentLoaded',lambda_function1,False)

def draw(v,c,bc,w,h,value):
    if (v.paused or v.ended):return False 
    js.bc.drawImage(v,0,0,w,h)
    idata = js.bc.getImageData(0,0,w,h)
    data = idata.data 
    for i in range(0,data.length,4):
        r = data[i]
        g = data[i+1]
        b = data[i+2]
        data[i] = r * vidData[0] * vidData[1] 
        data[i+1] = g * vidData[0] * vidData[2] 
        data[i+2] = b * vidData[0] * vidData[3]
    idata.data = data

    js.c.putImageData(idata,0,0)
    js.c.putImageData(idata,0,0) 
    def lambda_function3():
        draw(v,c,bc,w,h,value) 
    drawTimeout = js.setTimeout(lambda_function3, 0)


def changeVideo(attribute,direction):
    vidData[attribute] += direction * .5
